client/src/components/NewProductForm.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react'
+import { hot } from 'react-hot-loader/root'
+import { Redirect } from 'react-router-dom'
+
+import ErrorList from './ErrorList'
+import translateServerErrors from '../services/translateServerErrors'
+
+const newProductForm = props => {
+  const [newProduct, setNewProduct] = useState({
+    productName: '',
+    brandName: '',
+    description: ''
+  })
+
+  const categoryId = props.match.params.categoryId
+  const [shouldRedirect, setShouldRedirect] = useState(false)
+  const [errors, setErrors] = useState({})
+
+  const handleInputChange = (event) => {
+    setNewProduct({
+      ...newProduct,
+      [event.currentTarget.name]: event.currentTarget.value
+    })
+  }
+
+  const addProduct = async (newProduct) => {
+    try {
+      const response = await fetch(`/api/v1/categories/${categoryId}/products`, {
+        method: 'POST',
+        headers: new Headers({
+          'Content-Type': 'application/json'
+        }),
+        body: JSON.stringify(newProduct)
+      })
+      if (!response.ok) {
+        if(response.status === 422) {
+          const body = await response.json()
+          const newErrors = translateServerErrors(body.errors)
+          return setErrors(newErrors)
+        } else {
+          const errorMessage=`${response.status} (${response.statusText})`
+          throw new Error(errorMessage)
+        }
+      } else {
+        const body = await response.json()
+        console.log(body)
+        setShouldRedirect(true)
+      }
+    } catch ( error ) {
+      console.error(`Error in fetch: ${error.message}`)
+    }
+  }
+
+  const clearForm = () => {
+    setNewProduct({
+      productName: '',
+      brandName: '',
+      description: ''
+    })
+    setErrors({})
+  }
+  
+  const handleSubmit = (event) => {
+    event.preventDefault()
+    addProduct(newProduct)
+    clearForm()
```

Comment:
no update needed BUT, this technically clears the form even if they messed up. It would be nice to not clear the form if there are validation errors. `addProduct` would need to return true or false though, and you would need an if...else statement

again, just a thought, no change required